File: clojure/perfect-numbers/src/perfect_numbers.clj
(ns perfect-numbers)

(defn factors-of
  [num]
  (->>
   (range 1 (inc (/ num 2)))
   (filter (comp zero? (partial rem num)))))

(defn classify
  [num]
  (if (neg? num)
    (throw (IllegalArgumentException. "Number shouldn't be negative."))
    (let [as (apply + (factors-of num))]
      (cond
        (< as num) :deficient
        (> as num) :abundant
        (= as num) :perfect))))

File: csharp/perfect-numbers/PerfectNumbers.cs
using System;
using System.Linq;

public enum Classification
{
    Perfect,
    Abundant,
    Deficient
}

public static class PerfectNumbers
{
    public static Classification Classify(int number)
    {
        if (number <= 0) throw new ArgumentOutOfRangeException();
        var aliquotSum = GetAliquotSum(number);
        var myValue = aliquotSum switch 
        {
            < number => Classification.Deficient,
            > number => Classification.Abundant,
            _ => Classification.Perfect
        };
        return myValue;
    }

    private static int GetAliquotSum(int number) => Enumerable.Range(1, number / 2).Where(factor => number % factor == 0).Sum();
}